crates/video/src/main.rs

```diff
@@ -0,0 +1,213 @@
+
+        let mut vring_workers = daemon.get_epoll_handlers();
+        for thread in vu_video_backend.read().unwrap().threads.iter() {
+            thread
+                .lock()
+                .unwrap()
+                .set_vring_workers(vring_workers.remove(0));
+        }
+
+        let listener =
+            Listener::new(socket.clone(), true).map_err(Error::FailedCreatingListener)?;
+        daemon.start(listener).expect("Stargin daemon");
+
+        match daemon.wait() {
+            Ok(()) => {
+                info!("Stopping cleanly");
+            }
+            Err(vhost_user_backend::Error::HandleRequest(vhost_user::Error::PartialMessage)) => {
+                info!(
+                    "vhost-user connection closed with partial message.
+                    If the VM is shutting down, this is expected behavior;
+                    otherwise, it might be a bug."
+                );
+            }
+            Err(e) => {
+                warn!("Error running daemon: {:?} -> {}", e, e.to_string());
+            }
+        }
+
+        vu_video_backend
+            .read()
+            .unwrap()
+            .exit_event
+            .write(1)
+            .expect("Shutting down worker thread");
+    }
+}
+
+fn main() -> Result<()> {
+    env_logger::init();
+
+    start_backend(VuVideoConfig::try_from(VideoArgs::parse()).unwrap())
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use assert_matches::assert_matches;
+    use rstest::*;
+    use tempfile::tempdir;
+
+    #[rstest]
+    // No device specified defaults to /dev/video0
+    #[case::no_device(vec!["", "-s  video.sock ", "-b", "null"],
+    VideoArgs {
+        socket_path: "video.sock".to_string(),
+        v4l2_device: "/dev/video0".to_string(),
+        backend: BackendType::Null,
+    })]
+    // Specifying device overwrite the default value
+    #[case::set_device(vec!["", "-s  video.sock ", "-d /dev/video1 ", "-b", "null"],
+    VideoArgs {
+        socket_path: "video.sock".to_string(),
+        v4l2_device: "/dev/video1".to_string(),
+        backend: BackendType::Null,
+    })]
+    // Selecting different decoder
+    #[case::set_v4l2_decoder(vec![" ", "--socket-path", "long-video.sock ", "-b", "v4l2-decoder"],
+    VideoArgs {
+        socket_path: "long-video.sock".to_string(),
+        v4l2_device: "/dev/video0".to_string(),
+        backend: BackendType::V4L2Decoder,
+    })]
+    fn test_command_line_arguments(#[case] args: Vec<&str>, #[case] command_line: VideoArgs) {
+        let args: VideoArgs = Parser::parse_from(args.as_slice());
+
+        assert_eq!(
+            VuVideoConfig::try_from(command_line).unwrap(),
+            VuVideoConfig::try_from(args).unwrap()
+        );
+    }
+
+    #[test]
+    fn test_fail_create_backend() {
+        use vhu_video::VuVideoError;
+        let config = VideoArgs {
+            socket_path: "video.sock".to_string(),
+            v4l2_device: "/path/invalid/video.dev".to_string(),
```

Comment:
```suggestion
            socket_path: "video.sock".to_owned(),
            v4l2_device: "/path/invalid/video.dev".to_owned(),
```
and everywhere else where &str -> String is converted. The semantics are clearer using that compared to ToString.